src/kernels/MIOpenSoftmax.cl

```diff
@@ -689,7 +689,12 @@ __kernel void SoftmaxBackward(global _FLOAT* y,
 {
 
 #if NUM_BATCH == 1 // CSR-Vector like appraoch
+
+#if USE_SOFTMAX_LOG && MIOPEN_USE_FP16
+    local float l_helper[256];
+#else
     local _FLOAT l_helper[256];
+#endif
```

Comment:
I would properly define `_FLOAT` type at the beginning and keep the rest code untouched, or introduce `FLOAT_ACCUM` define for all the accumulators.

Reply:
I agree that the code would be more clear if it defined _FLOAT and _FLOAT_ACCUM and use the accum type when the extra precision is desired.